WMF Framework/NSCharacterSet+WMFLinkParsing.m

```diff
@@ -7,7 +7,7 @@ + (NSCharacterSet *)wmf_URLArticleTitlePathComponentAllowedCharacterSet {
     static NSCharacterSet *wmf_URLArticleTitleAllowedCharacterSet;
     dispatch_once(&onceToken, ^{
         NSMutableCharacterSet *pathAllowedCharacterSet = [[NSCharacterSet URLPathAllowedCharacterSet] mutableCopy];
-        [pathAllowedCharacterSet removeCharactersInString:@"/."];
```

Comment:
It's a bit hard to track down what all could be affected but on first glance it seems like continue reading and related pages content group URLs that are generated hit this. Seems like something like this snippet from `WMFContinueReadingContentSource.m` could create a different `continueReadingURL` than what is persisted, in which case this fetch would start to return nothing.

```
NSURL *continueReadingURL = [WMFContentGroup continueReadingContentGroupURLForArticleURL:lastRead];
[moc fetchOrCreateGroupForURL:continueReadingURL
        ofKind:WMFContentGroupKindContinueReading
        forDate:userData.viewedDate
        withSiteURL:nil
        associatedContent:nil
        customizationBlock:^(WMFContentGroup *_Nonnull group) {
                group.contentPreview = lastRead;
        }
];
```

Reply:
Good catch - for continue reading the old content sources are removed but for related pages, this change could cause a duplicate entry in the feed for articles with + in the title. I also found that the talk page (and anywhere else we construct a URL by setting wmf_title) now have the + encoded, but that is working fine.

Reply:
For the duplicate related page issue, I'd rather take the risk that users with a related page card for an article with a + in the title will see a duplicate related page card instead of adding a migration to re-encode content group keys that will slow down migration for all users, but it's worth flagging the tradeoff to the rest of the team to weigh in.

Reply:
@joewalsh to clarify, this would just be in the transition period (based on what they have in the existing app version, and what this would provide)? So users wouldn't be seeing duplicates in perpetuity? (This is my first time looking at that corner of the codebase.)

Reply:
@mcleinman just in the transition period